internal/service/advancedcluster/model_sdk_version_conversion.go

```diff
@@ -0,0 +1,130 @@
+package advancedcluster
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	admin20231115 "go.mongodb.org/atlas-sdk/v20231115014/admin"
+	"go.mongodb.org/atlas-sdk/v20240530001/admin"
+)
+
+// Conversions from one SDK model version to another are used to avoid duplicating our flatten/expand conversion functions.
+// - These functions must not contain any business logic.
+// - All will be removed once we rely on a single API version.
+
+func convertTagsToLatest(tags *[]admin20231115.ResourceTag) *[]admin.ResourceTag {
```

Comment:
we might want a companion test file with unit tests

Reply:
had my doubts on value of tests in this case as there is no business logic other than passing values from one model to another